addon/ExperimentProvider.js

```diff
@@ -1,60 +1,101 @@
-const seedrandom = require("common/vendor")("seedrandom");
-const simplePrefs = require("sdk/simple-prefs");
-const OVERRIDE_PREF = "experimentOverrides";
+const prefService = require("sdk/preferences/service");
+const {PrefsTarget} = require("sdk/preferences/event-target");
+const ss = require("sdk/simple-storage");
 
-exports.OVERRIDE_PREF = OVERRIDE_PREF;
 exports.ExperimentProvider = class ExperimentProvider {
-  constructor(clientID, experiments = require("../experiments.json"), rng) {
-    this._clientID = clientID;
+  constructor(experiments = require("../experiments.json"), rng) {
     this._experiments = experiments;
-    this._rng = rng || seedrandom(clientID);
+    this._rng = rng || Math.random;
     this._data = {};
     this._experimentId = null;
+    ss.storage.overrideExperimentProvider = false;
+    this._target = PrefsTarget();
   }
 
   init() {
     this.setValues();
-    this._onPrefChange = () => this.setValues();
-    simplePrefs.on(OVERRIDE_PREF, this._onPrefChange);
+    this._onPrefChange = prefName => this.overrideExperimentPrefs(prefName);
+
+    Object.keys(this._experiments).forEach(experimentName => {
+      this._target.on(experimentName, this._onPrefChange);
+      Object.defineProperty(this._data, experimentName, {
+        get() {
+          return prefService.get(experimentName);
+        }
+      });
+    });
+  }
+
+  overrideExperimentPrefs(prefName) {
+    ss.storage.overrideExperimentProvider = true;
+    this._experimentId = null;
   }
 
   setValues() {
-    // Load override pref, if defined
-    const overrides = simplePrefs.prefs[OVERRIDE_PREF] && simplePrefs.prefs[OVERRIDE_PREF].split(/,?\s+/);
+    if (ss.storage.overrideExperimentProvider) {
+      console.log(`The following experiments were turned on via overrides:\n`); // eslint-disable-line no-console
+      for (let experimentName in this._experiments) {
+        if (prefService.has(experimentName)) {
+          console.log(`- ${experimentName} - \n`); // eslint-disable-line no-console
+        }
+      }
+      return;
+    }
 
     const randomNumber = this._rng();
     let floor = 0;
+    let inExperiment;
+
+    Object.keys(this._experiments).forEach(key => {
+      const experiment = this._experiments[key];
+      const {variant} = experiment;
+
+      // If the user is part of an inactive experiment,
+      // reset that experiment's pref.
+      if (prefService.get(key) && !experiment.active) {
+        prefService.set(key, false);
+      }
+
+      // If the user is already part of an active experiment, set the experiment id.
+      if (prefService.get(key) && experiment.active) {
```

Comment:
In both if statements here, if the `variant` value of the experiment is falsey (e.g. `0`), this will incorrectly resolve to `false`; don't you want to check `undefined`?